aziotctl/src/init.rs

```diff
@@ -423,6 +423,7 @@ fn run_inner(stdin: &mut impl Reader) -> Result<RunOutput> {
             homedir_path: AZIOT_CERTD_HOMEDIR_PATH.into(),
             cert_issuance: Default::default(),
             preloaded_certs: Default::default(),
+            principal: Default::default(),
```

Comment:
This will need to give IS access to the device ID cert in case `device_id_source` is `Some()`.

Add a `if device_id_source.is_some() { certd_config.principal.push(...); }` at the end of this block, right before it returns `certd_config` (line 636).

Reply:
Does IS need to modify the device ID cert? Reading the cert is authorized for everyone.

Reply:
I thought dynamically issued certs (requesting a cert that doesn't exist but has an issuance defined, so it's created as part of getting it) would require authorization equivalent to creating them, but I see now you only check the auth for import and delete.

I guess it's fine.

Reply:
Do you know how to fix the error:
```
could not serialize aziot-certd config

Caused by:
    values must be emitted before tables', aziotctl/src/init.rs:1161:46
```

I tried a few things that didn't work. To repro: `cargo test -p aziotctl`

Reply:
https://news.ycombinator.com/item?id=24853386

Example of a workaround for the same reason: https://github.com/Azure/iot-identity-service/blob/8c471e53a8c77276f0af7e4f0d0e5eed06d57eed/identity/aziot-identityd-config/src/lib.rs#L90-L96

Reply:
Should be fixed now. Thanks.

Reply:
Makes sense. The serializer can't resume the struct after a struct-array member's `[[member]]` section.